### PR TITLE
Rustfmt everything

### PR DESCRIPTION
**Pull request by Manishearth**

### CODE REVIEW

File: src/entry.rs [hunk 1]
```diff
@@ -48,7 +48,11 @@ impl LateLintPass for HashMapLint {
                     // in case of `if !m.contains_key(&k) { m.insert(k, v); }`
                     // we can give a better error message
                     let sole_expr = else_block.is_none() &&
-                        if then_block.expr.is_some() { 1 } else { 0 } + then_block.stmts.len() == 1;
+                                    if then_block.expr.is_some() {
+                        1
```
Comment (mcarton) [inline src/entry.rs]: That’s weird.

Comment (Manishearth) [inline src/entry.rs]: yeah, just noticed that too. Fixed the code.

File: src/loops.rs [hunk 2]
```diff
@@ -748,14 +746,10 @@ fn is_ref_iterable_type(cx: &LateContext, e: &Expr) -> bool {
...
-    match_type(cx, ty, &HASHMAP_PATH) ||
-    match_type(cx, ty, &["std", "collections", "hash", "set", "HashSet"]) ||
+    is_iterable_array(ty) || match_type(cx, ty, &VEC_PATH) || match_type(cx, ty, &LL_PATH) ||
+    match_type(cx, ty, &HASHMAP_PATH) || match_type(cx, ty, &["std", "collections", "hash", "set", "HashSet"]) ||
     match_type(cx, ty, &["collections", "vec_deque", "VecDeque"]) ||
-    match_type(cx, ty, &["collections", "binary_heap", "BinaryHeap"]) ||
-    match_type(cx, ty, &BTREEMAP_PATH) ||
+    match_type(cx, ty, &["collections", "binary_heap", "BinaryHeap"]) || match_type(cx, ty, &BTREEMAP_PATH) ||
     match_type(cx, ty, &["collections", "btree", "set", "BTreeSet"])
 }
```
Comment (mcarton) [inline src/loops.rs]: I prefer the version with one match per line.

File: src/matches.rs [hunk 3]
```diff
@@ -135,9 +135,8 @@ impl LateLintPass for MatchPass {
 }
 
 fn check_single_match(cx: &LateContext, ex: &Expr, arms: &[Arm], expr: &Expr) {
-    if arms.len() == 2 &&
-       arms[0].pats.len() == 1 && arms[0].guard.is_none() &&
-       arms[1].pats.len() == 1 && arms[1].guard.is_none() {
+    if arms.len() == 2 && arms[0].pats.len() == 1 && arms[0].guard.is_none() && arms[1].pats.len() == 1 &&
+       arms[1].guard.is_none() {
         let els = if is_unit_expr(&arms[1].body) {
```
Comment (mcarton) [inline src/matches.rs]: I prefer the previous version, with `arms[0]` on one line and `arms[1]` on the other.

File: src/print.rs [hunk 4]
```diff
@@ -75,8 +74,10 @@ impl LateLintPass for PrintLint {
 fn is_in_debug_impl(cx: &LateContext, expr: &Expr) -> bool {
     let map = &cx.tcx.map;
 
-    if let Some(NodeImplItem(item)) = map.find(map.get_parent(expr.id)) { // `fmt` method
-        if let Some(NodeItem(item)) = map.find(map.get_parent(item.id)) { // `Debug` impl
+    if let Some(NodeImplItem(item)) = map.find(map.get_parent(expr.id)) {
+        // `fmt` method
+        if let Some(NodeItem(item)) = map.find(map.get_parent(item.id)) {
+            // `Debug` impl
             if let ItemImpl(_, _, _, Some(ref tr), _, _) = item.node {
```
Comment (mcarton) [inline src/print.rs]: The comment is now at the wrong place, IMO it should be on the same line as the  corresponding `if let` or on the previous line.

**Pull request merged**